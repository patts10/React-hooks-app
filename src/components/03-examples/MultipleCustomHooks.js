import React from 'react';
import { useCounter } from '../../hooks/useCounter';
import { useFetch } from '../../hooks/useFetch';
import '../02-useEffect/effects.css'

export const MultipleCustomHooks = () => {

  const { counter, increment } = useCounter(1);

  const { loading, data } = useFetch(`https://www.breakingbadapi.com/api/quotes/${ counter }`);

  const {author, quote } = (!loading && data.length>0) && data[0];

  return (
    <>
      <h1 className='font-semibold mb-3'>Breaking Quotes</h1>
      <hr />

      {
        loading
        ?
          (
      <div className='bg-sky-200 bg-opacity-75 py-3 my-3 text-center'>
        Loading...
      </div>

          )
        :
            (
      <blockquote className='text-right my-3'>
        <p className='mb-0 font-bold'>{ quote }</p>
        <footer className='blockquote-footer'>{ author }</footer>
      </blockquote>

            )
      }

      <button className='bg-blue-500 shadow-lg shadow-blue-500/50 rounded py-1 px-2 mt-3 text-white'
        onClick={ increment }
      >
        Next quote
      </button>


    </>
  )
}
