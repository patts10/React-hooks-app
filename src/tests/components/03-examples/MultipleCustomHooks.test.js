
import { shallow } from 'enzyme'
import { MultipleCustomHooks } from '../../../components/03-examples/MultipleCustomHooks';
import { useFetch } from '../../../hooks/useFetch';
import { useCounter } from '../../../hooks/useCounter';
jest.mock('../../../hooks/useFetch');
jest.mock('../../../hooks/useCounter');

describe('Tests on <MultipleCustomHooks />', () => { 

  beforeEach( () => {
    useCounter.mockReturnValue( {
      counter: 10,
      increment: () => {}
    });
  });
  
  test('should show component correctly', () => { 

    useFetch.mockReturnValue({
      data: null,
      loading: true,
      error: null
    })
    
    const wrapper = shallow(<MultipleCustomHooks />);
    expect(wrapper).toMatchSnapshot();
   });

   test('should show info', () => { 
     
    useFetch.mockReturnValue( {
      data: [{
        author: 'Jona',
        quote: 'Progress is never linear'
      }],
      loading: false,
      error: null
    });

    const wrapper = shallow( <MultipleCustomHooks /> );

    expect(wrapper.find('bg-sky-200').exists()).toBe( false );
    expect(wrapper.find('.mb-0').text().trim()).toBe('Progress is never linear');
    expect(wrapper.find('footer').text().trim()).toBe('Jona');

    });
 });