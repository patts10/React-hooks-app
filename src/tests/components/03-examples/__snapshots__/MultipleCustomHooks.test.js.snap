// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <MultipleCustomHooks /> should show component correctly 1`] = `
<Fragment>
  <h1
    className="font-semibold mb-3"
  >
    Breaking Quotes
  </h1>
  <hr />
  <div
    className="bg-sky-200 bg-opacity-75 py-3 my-3 text-center"
  >
    Loading...
  </div>
  <button
    className="bg-blue-500 shadow-lg shadow-blue-500/50 rounded py-1 px-2 mt-3 text-white"
    onClick={[Function]}
  >
    Next quote
  </button>
</Fragment>
`;
